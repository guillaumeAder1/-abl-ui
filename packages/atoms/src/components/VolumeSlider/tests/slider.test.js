import { render, getByText } from '@testing-library/react';
import React from 'react';
import { Slider } from '../';

describe('Slider Component', () => {
	test('should render', () => {
		const { container, debug } = render(<Slider />);
		expect(container).toMatchSnapshot();
		getByText(container, '100');
		getByText(container, '0');
	});
});
