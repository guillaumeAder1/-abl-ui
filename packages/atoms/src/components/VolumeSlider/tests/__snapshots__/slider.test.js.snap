// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component should render 1`] = `
<div>
  <style>
    
	.slider {
		display: flex;
		height: 130px;
		position:relative;
		width: 40px;
		margin: auto;
		justify-content: space-between;
	}
	.slider:focus {
		outline: 1px dashed black;
		padding: 0 5px;
	}
	.section {
		flex: 1 1 35%;
		display: flex;
		justify-content: space-between;
		margin: 5px 0; 
	}
	.bar{
		height: 100%;
		flex-basis: 47%;
		background-color: black;
	}
	.scale{
		width: 5px;
		display: flex;
		justify-content: space-between;
		flex-direction: column;
		margin: 5px 0; 
	}
	.tick {
		margin-left: 2px;
		height: 1px;
		background: black;
	}
	.tick.short{
		width: 60%;
	}
	.tick.long{
		width: 100%;
	}
	.number{
		flex-direction: column;
		font-size: 10px;
		line-height: 10px;
		font-family: helvetica;
		text-align: end;
	}

  </style>
  <div
    aria-label="slider"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="90"
    class="slider"
    role="slider"
    tabindex="0"
  >
    <div
      class="section"
    >
      <div
        class="bar"
      />
      <div
        class="bar"
      />
    </div>
    <div
      class="scale"
    >
      <div
        class="css-1ur5jav-basePointer-Point e1utnk6q1"
      />
      <div
        class="tick long"
      />
      <div
        class="tick short"
      />
      <div
        class="tick short"
      />
      <div
        class="tick long"
      />
      <div
        class="tick short"
      />
    </div>
    <div
      class="section number"
    >
      <div>
        100
      </div>
      <div>
        0
      </div>
    </div>
  </div>
</div>
`;
